### PR TITLE
New reference architecture diagram: Protect data center networks

### PR DESCRIPTION
**Pull request by securitypedant**
### Documentation checklist

- [ ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [ ] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.
- [ ] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+- [Magic Transit](https://www.cloudflare.com/network-services/products/magic-transit/) provides instant detection and mitigation against network-layer DDoS attacks on your public, Internet-facing networks.
+- [Magic WAN](https://www.cloudflare.com/network-services/products/magic-wan/) provides any-to-any, hybrid/multi-cloud secure connectivity between your private, enterprise networks.
+- [Magic Firewall](/magic-firewall/) is a cloud-native network firewall service that can be used to filter traffic that is routed to and from your networks that are protected by Magic Transit. It also supports functionalities such as [Intrusion Detection](/magic-firewall/about/ids/) (IDS) and [packet capture](/magic-firewall/packet-captures/).
```
Comment (marciocloudflare) [inline src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx:27]: Should we add Magic WAN here as well? MFirewall is also available to customers who purchase MWAN.

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+- [Gateway](https://www.cloudflare.com/zero-trust/products/gateway/) is a secure web gateway (SWG) service that allows you to inspect and control both Internet bound traffic that is originated from your networks, as well as private network-to-private network traffic (that is, east-west), by proxying such traffic through Cloudflare's global network while applying DNS, network and HTTP based [policies](/cloudflare-one/policies/gateway/).
```
Comment (marciocloudflare) [inline src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]:
```suggestion
- [Gateway](https://www.cloudflare.com/zero-trust/products/gateway/) is a secure web gateway (SWG) service that allows you to inspect and control both Internet-bound traffic that is originated from your networks, as well as private network-to-private network traffic (that is, east-west), by proxying such traffic through Cloudflare's global network while applying DNS, network and HTTP based [policies](/cloudflare-one/policies/gateway/).
```

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+This document focuses specifically on the reference architectures of using Cloudflare Magic Transit, Magic WAN, Magic Firewall and Cloudflare Gateway services to protect both external and internal communications to your data center networks. For details of how Magic Transit, Magic WAN, Magic Firewall and Cloudflare Gateway works and how it can be architected for various use cases, see the linked resources at the end of the document.
+
+To illustrate the architecture and how it works, the following diagrams visualize an example corporation with a set of data center networks that are either public-facing, connecting to users on the Internet or private, internal facing, used for communication within the enterprise. These networks are deployed at two on-premises locations. The prefixes of the public-facing networks are to be protected by Cloudflare Magic Transit.
+
+| Data center 1                       | Data center 2                         |
+| :---------------------------------- | :------------------------------------ |
+| Public-facing network: 192.0.2.0/24 | Public-facing network: 203.0.113.0/24 |
+| Private network: 192.168.1.0/24     | Private network: 172.16.2.0/24        |
```
Comment (marciocloudflare) [inline src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]:
```suggestion
| Public-facing network: `192.0.2.0/24` | Public-facing network: `203.0.113.0/24` |
| Private network: `192.168.1.0/24`     | Private network: `172.16.2.0/24`        |
```

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+The edge router(s) at each data center is connected to Cloudflare network via two Direct [Cloudflare Network Interconnect](/network-interconnect/) (CNI) connections, which are direct, private connections between your network and Cloudflare network, one for carrying public-facing network traffic, one for carrying private network traffic. Optionally, you can choose to carry both public and private network traffic over a single CNI connection but many organizations do desire to transport external and internal network traffic over separate connections in their security practice.
```
Comment (marciocloudflare) [inline src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]:
```suggestion
The edge router(s) at each data center is connected to Cloudflare network via two Direct [Cloudflare Network Interconnect](/network-interconnect/) (CNI) connections, which are direct, private connections between your network and Cloudflare network. One of the Direct CNI connections is for carrying public-facing network traffic, while the other is for carrying private network traffic. Optionally, you can choose to carry both public and private network traffic over a single CNI connection but many organizations do desire to transport external and internal network traffic over separate connections in their security practice.
```

Comment (marciocloudflare) [reply src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]: Cut sentence length to help legibility.

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+4. Once traffic clears inspection, Gateway proxies the outbound traffic to their destinations on the Internet. The source IP addresses of the outbound traffic are the Cloudflare owned IP addresses associated with the Gateway service, which if you want you can purchase and set your [own egress IP](/cloudflare-one/policies/gateway/egress-policies/dedicated-egress-ips/)
+5. Return traffic from the Internet, destined to Cloudflare's IP addresses linked to the Gateway service, is routed into Cloudflare's global network.
+6. Traffic is inspected against Gateway policies.
+7. Return traffic that passes Gateway inspection is routed to Magic Firewall for further packet filtering.
+8. Return traffic that passes Magic Firewall filtering is routed from Cloudflare to your network locations via CNIs that transport public-facing network traffic.
+
+## Protect site-to-site, inter-data center, private network traffic
+
+The reference architecture diagrams below illustrate how Cloudflare services \- Magic WAN, Magic Firewall and Cloudflare Gateway, can be used to protect site-to-site, inter-data center traffic between your private networks.
```
Comment (marciocloudflare) [inline src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]:
```suggestion
The reference architecture diagrams below illustrate how Cloudflare services — Magic WAN, Magic Firewall and Cloudflare Gateway — can be used to protect site-to-site, inter-data center traffic between your private networks.
```

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+**Site to Site Private Network Traffic Connectivity**
+First, let us look at the use case where you do not intend to subject site-to-site private network traffic to Cloudflare Gateway proxy firewall service and simply route it using Magic WAN service.
```
Comment (marciocloudflare) [inline src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]:
```suggestion
**Site to Site Private Network Traffic Connectivity**

First, let us examine the use case where you do not intend to subject site-to-site private network traffic to Cloudflare Gateway proxy firewall service and simply route it using Magic WAN service.
```

Comment (marciocloudflare) [reply src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]: Changed verb for inclusivity.

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+1. Each site routes site-to-site private network traffic, destined to the other data center location, to Cloudflare Magic WAN via the corresponding CNI connections. This can be done at your site through routing techniques of your choice, such as policy based routing (PBR).
+2. Upon entering the Cloudflare network, traffic is routed through Magic Firewall.
+3. Magic Firewall subjects traffic to any configured network firewall policies.
+4. Traffic that clears the Magic Firewall rules and is not intended to be further proxied by Cloudflare Gateway service, is routed back to the destination network via the corresponding CNI.
+
+**Site to Site Private Network Traffic with Application Level Security Controls**
+For the use case where you do want to apply application level policy for fine-grain control and security on certain private network traffic, you can route and proxy such traffic through Magic WAN and Cloudflare Gateway service. The following diagram illustrates the architecture and packet flow of such use cases.
```
Comment (marciocloudflare) [inline src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]:
```suggestion
**Site to Site Private Network Traffic with Application Level Security Controls**

For the use case where you do want to apply application level policy for fine-grain control and security on certain private network traffic, you can route and proxy such traffic through Magic WAN and Cloudflare Gateway service. The following diagram illustrates the architecture and packet flow of such use cases.
```

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+5. The proxied traffic, en-route to its destination private network, is routed through Magic Firewall once again for further packet filtering.
+6. Traffic that passes Magic Firewall filtering is routed from Cloudflare to your network locations via the corresponding CNIs that transport private network traffic.
+
+## Protect outbound Internet traffic from private networks
+
+The reference architecture diagram below illustrates how Cloudflare services \- Magic WAN, Magic Firewall and Cloudflare Gateway, can be used to protect outbound Internet traffic originating from the data centers' private networks. The use cases and the protection provided to the servers on the private networks are very similar to those described in the previous section about protecting Internet access from public-facing networks. The differences are that the servers have private IP addresses and that Magic WAN service is used in this section, as opposed to the previous section where servers are assigned with public IP addresses and Magic Transit server is used.
```
Comment (marciocloudflare) [inline src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx]:
```suggestion
The reference architecture diagram below illustrates how Cloudflare services — Magic WAN, Magic Firewall and Cloudflare Gateway — can be used to protect outbound Internet traffic originating from the data centers' private networks. The use cases and the protection provided to the servers on the private networks are very similar to those described in the previous section about protecting Internet access from public-facing networks. The differences are that the servers have private IP addresses and that Magic WAN service is used in this section, as opposed to the previous section where servers are assigned with public IP addresses and Magic Transit server is used.
```

**Review by marciocloudflare — APPROVED** [review]

**Comment by marciocloudflare** [comment]
Approved with a few suggestions @securitypedant 
Thanks!

File: src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx (hunk 1 above)
Comment (securitypedant) [reply src/content/docs/reference-architecture/diagrams/network/protect-data-center-networks.mdx:27]: Magic WAN is already in the list.

**Pull request merged**